### PR TITLE
Kv lsp enabled flag

### PR DESCRIPTION
**Pull request by qinqon**
#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: go-controller/pkg/ovn/base_network_controller_pods.go [hunk 1]
```diff
@@ -1178,6 +1187,6 @@ func (bnc *BaseNetworkController) wasPodReleasedBeforeStartup(uid, nad string) b
 	return bnc.releasedPodsBeforeStartup[nad].Has(uid)
 }
 
-func isAllowedForMigration(isSecondary, isPrimaryNetwork, isl2Topology bool) bool {
-	return isSecondary && !isPrimaryNetwork && isl2Topology
+func (bnc *BaseNetworkController) isAllowedForMigration() bool {
```
Comment (maiqueb) [inline go-controller/pkg/ovn/base_network_controller_pods.go:1190]: I'd have this as a function instead. There's no need for the layer3 controller to have this method for instance.

Also, should't localnet be also a "customer" of this feature ?... The exact same mechanism should be re-used there to improve the migration downtime, right ?

File: go-controller/pkg/kubevirt/pod.go [hunk 2]
```diff
@@ -402,19 +376,56 @@ func filterNotComplete(vmPods []*corev1.Pod) []*corev1.Pod {
...
+
+	// Sort vmPods by creation time
+	sort.Slice(vmPods, func(i, j int) bool {
+		// i less than j
+		return vmPods[j].CreationTimestamp.After(vmPods[i].CreationTimestamp.Time)
+	})
+
+	status := LiveMigrationStatus{
+		SourcePod: vmPods[0],
+		TargetPod: vmPods[1],
```
Comment (maiqueb) [inline go-controller/pkg/kubevirt/pod.go:423]: could we trim down the objects ? IIUC we just require the labels on the pods. Right ?

File: go-controller/pkg/kubevirt/pod.go [hunk 3]
```diff
@@ -402,19 +376,56 @@ func filterNotComplete(vmPods []*corev1.Pod) []*corev1.Pod {
...
-		return false, nil
+	// no migration
+	if len(vmPods) < 2 {
+		return nil, nil
 	}
 
-	if isSourcePodAlreadyStale(pod, vmPods) && isTargetPodReady(getTargetPod(vmPods)) {
-		return true, nil
+	if len(vmPods) > 2 {
+		return nil, fmt.Errorf("unexpected live migration state at pods: %+v", vmPods)
```
Comment (maiqueb) [inline go-controller/pkg/kubevirt/pod.go:412]: too much info IMHO

File: go-controller/pkg/ovn/base_network_controller_pods.go (hunk 1 above)
Comment (qinqon) [reply go-controller/pkg/ovn/base_network_controller_pods.go:1190]: Having a function that is going to receive NetInfo is kind of the same also if will be in the same context as layer3, also I am total agains a free function receiving random booleans.

Comment (maiqueb) [reply go-controller/pkg/ovn/base_network_controller_pods.go:1190]: I don't understand what you mean, but FWIW, I also don't want a function that receives random booleans.

**Pull request closed without merging**